### DISCUSSION TITLE
Disk cache duration ?

### DISCUSSION THREAD

**Discussion by jscti** [coil-kt/coil#1248] [General]
Didn't find much details about the cache implementation/configuration of the cache policies.

I have a simple goal : cache images to disk for a long time. Example : country flags fetched through api/network or weather icons.
As this icons are basically static/constant, It would be totally fine if it was cached for a month/year/ or even lifetime, but it seems they are only cached for 24h (my guess) (?)

Am i right ? And is there any way to customize the cache duration somehow ?

Using it with : 

```
Image(
        painter = rememberImagePainter(race.getCountryFlagUri(),
            builder = {
                crossfade(true)
                placeholder(R.drawable.loading_img)
                error(R.drawable.ic_broken_image)
                diskCachePolicy(CachePolicy.ENABLED)
            }
        ),
        contentDescription = race.countryName,
    )
```

Thanks

**Comment by colinrtwhite** [comment]
It's likely the server is setting a custom `Cache-Control` header that restricts how long the image is cached. You can create an interceptor that overwrites that header. In 2.x you can also set `ImageLoader.Builder.respectCacheHeaders(false)` to disable cache headers.

**Comment by jscti** [comment]
Thanks for your reply.

I migrated to coil 2.

Lifetime cache is OK with `respectCacheHeaders(false)` 👍 

Now, if I want to ignore server cache and force it (on my side) to keep it cached X seconds on DISK (tested with 30s), nothing I tried worked (= cache is still loaded from memory.
Tried interceptor and per-request (as in https://github.com/coil-kt/coil/blob/main/docs/recipes.md#using-a-custom-okhttpclient) but failed.

Code :
```
 AsyncImage(
        model = ImageRequest.Builder(LocalContext.current)
            .data(data)
            .crossfade(true)
            .setHeader("Cache-Control", "max-age=30")
            .removeHeader("Expires")
            .build()
)
```

Image I'm loading is from github : 
url: https://raw.githubusercontent.com/jscti/json_nextgp/main/icons/dark/10d.png
response headers (from chrome but also tried in curl by tweaking cache request headers) : 
```
accept-ranges: bytes
access-control-allow-origin: *
cache-control: max-age=300
content-length: 825
content-security-policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
content-type: image/png
date: Tue, 26 Apr 2022 14:22:35 GMT
etag: W/"fce709a19677ca4c0e2df4ac5aeb09557d80ba9123a3a8ee8877568d281d538e"
expires: Tue, 26 Apr 2022 14:27:35 GMT
source-age: 25
strict-transport-security: max-age=31536000
vary: Authorization,Accept-Encoding,Origin
via: 1.1 varnish
x-cache: HIT
x-cache-hits: 1
x-content-type-options: nosniff
x-fastly-request-id: df26c3b1243a15283f5d3d944e543844eb4aed67
x-frame-options: deny
x-github-request-id: 2E0C:DBD1:B9C238:CA5EF2:6267E04D
x-served-by: cache-cdg20749-CDG
x-timer: S1650982955.169947,VS0,VE1
x-xss-protection: 1; mode=block
```

And in fact .. when I don't use  `respectCacheHeaders(false)`, it seems there is never DISK cache, only memory.

Well I'm lost :D

Thanks
